src/vehicle_gui.cpp

```diff
@@ -2977,6 +2989,17 @@ struct VehicleViewWindow : Window {
 					ShowOrdersWindow(v);
 				}
 				break;
+			case WID_VV_HONK_HORN: // honk horn
+				switch (v->type) {
+					case VEH_TRAIN:
+						v->PlayLeaveStationSound();
+						break;
+					case VEH_SHIP:
+						PlayShipSound(v);
```

Comment:
Isn't this the same as `v->PlayLeaveStationSound`?
I.e. no need to make `PlayShipSound` external, and no need to switch on vehicle type.